ui/src/main/java/org/pentaho/di/ui/trans/step/BaseStepDialog.java

```diff
@@ -1552,4 +1557,16 @@ void showErrorDialog( Exception e ) {
               "BaseStep.Exception.UnexpectedErrorEditingConnection.DialogMessage" ), e );
     }
   }
+
+  /* Hack for testing to prevent circular dependency */
+  public static boolean isJUnitTest() {
```

Comment:
I'm hesitate on this change to BaseStepDialog as it's a core class to implement a check like this. What exactly causing the circular dependency?

Reply:
Yep, I tried just about everything I could think of.  The problem is that when JUnit runs this class some of the checks fail because now we require a user with the correct privileges to be logged in or we need to instrument the   plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/AbsSecurityProvider.java validateAction() to return without throwing an exception.  When I tried to mock this class IntelliJ complained about circular references.

Reply:
<img width="456" alt="Screen Shot 2020-04-02 at 2 17 40 PM" src="https://user-images.githubusercontent.com/864379/78284102-dc93d080-74ec-11ea-8421-636341efca0e.png">